src/Gii.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Yiisoft\Yii\Gii;
+
+use Psr\Container\ContainerInterface;
+use Yiisoft\Factory\Exceptions\NotFoundException;
+use Yiisoft\Yii\Gii\Exception\GeneratorNotFoundException;
+use Yiisoft\Yii\Gii\Generators\Controller\Generator;
+
+final class Gii implements GiiInterface
+{
+    private ContainerInterface $container;
+    private iterable $generators;
+
+    public function __construct(iterable $generators, ContainerInterface $container)
+    {
+        $this->generators = $generators;
+        $this->container  = $container;
+    }
+
+    public function addGenerator(string $name, $generator): void
+    {
+        $this->generators[$name] = $generator;
+    }
+
+    /**
+     * @param  string  $name
+     * @return GeneratorInterface
+     * @throws NotFoundException
+     * @throws GeneratorNotFoundException
+     */
+    public function getGenerator(string $name): GeneratorInterface
```

Comment:
Can we pre-cache already initialized generators into private property (e.g. `$generatorsInstances`)?

Reply:
Let's do it in the next PRs